pkg/services/librarypanels/database_test.go

```diff
@@ -0,0 +1,82 @@
+package librarypanels
+
+import (
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddLibraryPanel(t *testing.T) {
+	t.Run("should fail if library panel already exists", func(t *testing.T) {
+		lps, user := setupTestEnv(t, models.ROLE_EDITOR)
+		command := addLibraryPanelCommand{
+			OrgId:        1,
+			FolderId:     1,
+			SignedInUser: &user,
+			Title:        "Text - Library Panel",
+			Model: []byte(`
+	{
+      "datasource": "${DS_GDEV-TESTDATA}",
+      "id": 1,
+      "title": "Text - Library Panel",
+      "type": "text"
+    }
+`),
+		}
+
+		noErr := lps.addLibraryPanel(&command)
+		require.NoError(t, noErr)
+
+		err := lps.addLibraryPanel(&command)
+		require.Error(t, err)
+	})
+}
+
+func setupTestEnv(t *testing.T, orgRole models.RoleType) (LibraryPanelService, models.SignedInUser) {
+	cfg := setting.NewCfg()
+	cfg.FeatureToggles = map[string]bool{"panelLibrary": true}
+
+	lps := LibraryPanelService{
+		SQLStore: nil,
+		Cfg:      cfg,
+	}
+
+	overrideServiceFunc := func(d registry.Descriptor) (*registry.Descriptor, bool) {
+		descriptor := registry.Descriptor{
+			Name:         "LibraryPanelService",
+			Instance:     &lps,
+			InitPriority: 0,
+		}
+
+		return &descriptor, true
+	}
+
+	registry.RegisterOverride(overrideServiceFunc)
```

Comment:
Why do you need to register the service? Don't see the need, and it's also not cleaned up after the test.

Reply:
Otherwise the feature toggle will not be found and the library_panel will not exist and the test will fail.

Reply:
Good point on cleaning up, any suggestions on how?

Reply:
I see now why it's needed, because the service adds a migration. I don't know why we haven't seen this pattern in tests before. The `registry` package currently lacks a way to de-register a service, I think you'll have to add a function that removes the override, and call it via `t.Cleanup`.

Reply:
I added ClearOverrides function to registry and cleanup to test.